mbedtls/tests/client_server.rs

```diff
@@ -81,29 +80,29 @@ fn client(
 
 fn server(
     mut conn: TcpStream,
-    min_minor: i32,
-    max_minor: i32,
-    exp_minor: Result<i32, ()>,
+    min_version: Version,
+    max_version: Version,
+    exp_version: Result<Version, ()>,
 ) -> TlsResult<()> {
```

Comment:
Why is the expected version passed as a result? It also seems to me like it is acting semantically as an `Option<Version>`. I see that it is like this since before this particular PR, but I'd still like to know.

Reply:
It is `None` if (given the client+server configurations) it is expected to be impossible to negotiate. F.ex. if client is TLS 1.0 only and server is 1.2 only, then the expected version of the session is not any of the values enumerated in `Version`.

Reply:
Sorry I misparsed your question. It should be an `Option`.

Reply:
Ah ok I understand